tests/src/test/scala/pureconfig/ConfigSourceSuite.scala

```diff
@@ -0,0 +1,205 @@
+    val defaults = ConfigSource.resources("conf/configSource/defaults.conf")
+    val overrides = ConfigSource.resources("conf/configSource/overrides.conf")
+    val main = ConfigSource.resources("conf/configSource/main.conf")
+
+    overrides.
+      withFallback(main).
+      withFallback(defaults).
+      at("my-service").load[MyService] shouldBe Right(MyService("example.com", 8081, true))
+
+    val nonExisting1 = ConfigSource.resources("nonExistingResource1")
+    val nonExisting2 = ConfigSource.resources("nonExistingResource2")
+
+    overrides.
+      withFallback(nonExisting1).
+      withFallback(main).
+      withFallback(defaults).
+      at("my-service").load[MyService] should matchPattern {
+        case Left(ConfigReaderFailures(CannotReadResource("nonExistingResource1", _), Nil)) =>
+      }
+
+    overrides.
+      withFallback(nonExisting1).
+      withFallback(main).
+      withFallback(nonExisting2).
+      withFallback(defaults).
+      at("my-service").load[MyService] should matchPattern {
+        case Left(ConfigReaderFailures(
+          CannotReadResource("nonExistingResource1", _),
+          CannotReadResource("nonExistingResource2", _) :: Nil)) =>
+      }
+  }
+
+  it should "allow chaining optional fallback sources" in {
+    val defaults = ConfigSource.resources("conf/configSource/defaults.conf")
+    val overrides = ConfigSource.resources("conf/configSource/overrides.conf")
+    val main = ConfigSource.resources("conf/configSource/main.conf")
+
+    overrides.optional.
+      withFallback(main.optional).
+      withFallback(defaults.optional).
+      at("my-service").load[MyService] shouldBe Right(MyService("example.com", 8081, true))
+
+    val nonExisting1 = ConfigSource.resources("nonExistingResource1")
+    val nonExisting2 = ConfigSource.resources("nonExistingResource2")
+
+    overrides.optional.
+      withFallback(nonExisting1.optional).
+      withFallback(main.optional).
+      withFallback(defaults.optional).
+      at("my-service").load[MyService] shouldBe Right(MyService("example.com", 8081, true))
+
+    overrides.optional.
+      withFallback(nonExisting1.optional).
+      withFallback(main.optional).
+      withFallback(nonExisting2.optional).
+      withFallback(defaults.optional).
+      at("my-service").load[MyService] shouldBe Right(MyService("example.com", 8081, true))
+  }
+
+  it should "defer config resolution until all fallbacks are merged" in {
+    case class Conf(host: String)
+    val resolve1 = ConfigSource.resources("conf/configSource/resolve1.conf")
+    val resolve2 = ConfigSource.resources("conf/configSource/resolve2.conf")
+
+    resolve1.withFallback(resolve2).load[Conf] shouldBe Right(Conf("myhost1.example.com"))
+  }
+
+  it should "fail safely when a substitution can't be resolved" in {
+    case class Conf(hostOverride: String)
+    val resolve1 = ConfigSource.resources("conf/configSource/resolve1.conf")
+
+    resolve1.load[Conf] should failLike {
```

Comment:
Can you also add an example showing that calling `value()` on `resolve1` would fail similarly?

Reply:
Done